examples/desktop-registration/main.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"time"
+
+	teleport "github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+)
+
+func main() {
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	// Create a Teleport API client, loading credentials
+	// from an existing tsh profile on disk.
+	//
+	// The user that these creds are bound to must have a role
+	// granting create and update permissions on the windows_desktop
+	// resource.
+	clt, err := teleport.New(ctx, teleport.Config{
+		Credentials: []teleport.Credentials{teleport.LoadProfile("", "")},
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer clt.Close()
+
+	t := time.NewTicker(3 * time.Minute)
```

Comment:
I was going for < 1/2, so that the desktop wouldn't expire if one iteration failed due to a network flake or something. With 1/3, it would take two consecutive failures.